onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -745,4 +748,14 @@ public void onLocationChanged(Location l) {
     public void postInvalidate() {
         // Do nothing - calling `this.postInvalidate()` causes a StackOverflowError
     }
+
+    /**
+     * mLastLocation is defined statically.
+     * If the BaseMapFragment was initialized before, the new  BaseMapFragment doesn't notifies
+     * the onFocusChanged listener (see L:742).
+     * resetLastLocation can clear previous mLastLocation data.
+     */
+    public void resetLastLocation(){
+        mLastLocation = null;
+    }
```

Comment:
@barbeau I also need to add some method for ```mLastLocation``` as we discussed last week. Since, ```mLastLocation``` defined statically, If I create a new BaseMapFragment it never notifies the  onFocusChanged listener (L:727):
```java
 public void onLocationChanged(Location l) {
boolean firstFix = false;
if (mLastLocation == null) { 
// mLastLocation is static, so this is a first fix across all BaseMapFragments
firstFix = true;
}
mLastLocation = l;
if (mListener != null) {
// Show real-time location on map
mListener.onLocationChanged(l);
}
if (firstFix) {
// If its our first location (across all BaseMapFragments), move the camera to it
setMyLocation(true, false);
//Notify the onFocusChangeListener when we get the first location
onFocusChanged(null,null,l);
}
}
```

Please let me know if you have further comments. Maybe, we can handle this as an option in the constructor.